src/main/scala/nl/knaw/dans/easy/bagindex/components/BagStoreAccess.scala

```diff
@@ -107,16 +110,18 @@ trait BagStoreAccess {
    */
   def traverse: Try[Stream[(BagId, Path)]] = {
     def traverseBagStore(baseDir: Path): Try[Stream[(BagId, Path)]] = {
+      trace(baseDir)
 
       // we assume all bags in a bagstore are at equal depth, so following one path is enough!
       def probeForPathDepth: Try[Int] = Try {
-
         @tailrec
         def probe(path: Path, length: Int, levels: Int): Int = {
           length match {
             case l if l > 0 =>
-              val p = resource.managed(Files.list(path)).acquireAndGet(_.findFirst.get)
-              probe(p, length - p.getFileName.toString.length, levels + 1)
+              managed(Files.list(path)).acquireAndGet(_.findFirst.asScala) match {
```

Comment:
this is where the bug occurred: `_.findFirst.get` throws an exception in case `Files.list(path)` returns an empty `java.util.stream.Stream`. Instead we now transform it to a Scala `Option` and pattern match on it.